test/cmd/builds.sh

```diff
@@ -91,21 +91,21 @@ frombuild=$(oc start-build --from-build="${started}")
 os::cmd::expect_success_and_text "oc describe build ${frombuild}" 'centos/ruby-22-centos7$'
 echo "start-build: ok"
 
-os::cmd::expect_success "oc cancel-build ${started} --dump-logs --restart | grep 'Restarted build ${started}.'"
+os::cmd::expect_success_and_text "oc cancel-build ${started} --dump-logs --restart" "Restarted build ${started}."
 os::cmd::expect_success 'oc delete all --all'
 os::cmd::expect_success 'oc process -f examples/sample-app/application-template-dockerbuild.json -l build=docker | oc create -f -'
-os::cmd::expect_success "tryuntil 'oc get build/ruby-sample-build-1'"
+os::cmd::try_until_success 'oc get build/ruby-sample-build-1'
 # Uses type/name resource syntax to cancel the build and check for proper message
-os::cmd::expect_success "oc cancel-build build/ruby-sample-build-1 | grep 'Build build/ruby-sample-build-1 was cancelled.'"
+os::cmd::expect_success_and_text 'oc cancel-build build/ruby-sample-build-1' 'Build build/ruby-sample-build-1 was cancelled.'
 # Make sure canceling already cancelled build returns proper message
-os::cmd::expect_success "tryuntil $(oc cancel-build build/ruby-sample-build-1 | grep 'A cancellation event was already triggered for the build build/ruby-sample-build-1.')"
+os::cmd::try_until_text 'oc cancel-build build/ruby-sample-build-1' 'A cancellation event was already triggered for the build ruby-sample-build-1.'
 os::cmd::expect_success 'oc delete all --all'
 
 # Make sure failed build returns proper message when cancelled
 os::cmd::expect_success 'oc create -f test/fixtures/failing-bc.json'
-os::cmd::expect_success "tryuntil 'oc get bc failing-build'"
+os::cmd::try_until_success 'oc get bc failing-build'
 os::cmd::expect_success 'oc start-build failing-build'
-os::cmd::expect_success "tryuntil $(oc get build failing-build-1 | grep Failed)"
-os::cmd::expect_success "tryuntil $(oc cancel-build build/failing-build-1 | grep 'A build can be cancelled only if it has new/pending/running status.')"
+os::cmd::try_until_text 'oc get build failing-build-1' 'Failed'
```

Comment:
This is causing me issues, the build goes from `New` to `Pending` with the server spitting out 
```
I1201 08:36:29.939274    8384 controller.go:81] Handling build cmd-builds/ruby-sample-build-1
I1201 08:36:29.942565    8384 factory.go:38] Retrying Build cmd-builds/ruby-sample-build-1 with error: the image stream cmd-builds/origin-ruby-sample:latest cannot be used as the output for build cmd-builds/ruby-sample-build-1 because the integrated Docker registry is not configured and no external registry was defined
```
How many times do we try until we mark this as failed? @bparees @csrwng